src/pages/AcceptInvitation/pages/FillRequest/components/FillRequestForm.tsx

```diff
@@ -0,0 +1,130 @@
+import { FormControl, Stack, StackProps } from '@mui/material'
+import startCase from 'lodash/startCase'
+import { HTMLAttributes, useCallback } from 'react'
+import { Controller } from 'react-hook-form'
+
+import { CredentialRequest, fillOrgGroupRequest, OrgGroupRequest } from '@/api/modules/orgs'
+import { parsedCredentialSchemaProperty } from '@/api/modules/zkp'
+import { ErrorHandler } from '@/helpers'
+import { useForm, useMetamaskZkpSnapContext } from '@/hooks'
+import { UiButton, UiTextField } from '@/ui'
+
+type Props = StackProps & {
+  formProps?: HTMLAttributes<HTMLFormElement>
+  credentialFields: parsedCredentialSchemaProperty[]
+  orgGroupRequest: OrgGroupRequest
+  onRequestFilled?: () => void
+}
+
+export default function RequestForm({
+  formProps,
+  credentialFields,
+  orgGroupRequest,
+  onRequestFilled,
+  ...rest
+}: Props) {
+  const { userDid } = useMetamaskZkpSnapContext()
+
+  const {
+    formState,
+    isFormDisabled,
+    handleSubmit,
+    disableForm,
+    enableForm,
+    control,
+    getErrorMessage,
+  } = useForm(
+    {
+      ...credentialFields?.reduce(
+        (acc, field) => ({
+          ...acc,
+          [field.key]: field.value || '',
+        }),
+        {} as Record<string, string>,
```

Comment:
from the optimization perspective it's better to use reduce with defining new obj property instead of creating new one on each iteration:

```js
array.reduce((acc, item) => {
  acc[item.key] = item.value
  return acc
}, {})
```